### PR TITLE
adding more errors to fix Concourse issue with unreliable cred mgrs

### PR DESCRIPTION
**Pull request by romankartgh**
this is a first step towards fixing https://github.com/concourse/concourse/issues/2542

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@ralekseenkov Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/retryhttp&pullRequestId=4)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=concourse/retryhttp&pullRequestId=4) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@ralekseenkov Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/retryhttp&pullRequestId=4)!

**Comment by romankartgh** [comment]
@vito since the code is looking for suffixes:

```
remote error: handshake failure -> handshake failure
net/http: TLS handshake timeout -> handshake timeout
connect: connection reset by peer -> connection reset by peer
```

File: retryer.go [hunk 1]
```diff
@@ -38,5 +38,8 @@ var defaultRetryableErrors = []error{
 	syscall.ETIMEDOUT,
 	errors.New("i/o timeout"),
 	errors.New("no such host"),
-	errors.New("remote error: handshake failure"),
+	errors.New("handshake failure"),
+	errors.New("handshake timeout"),
+	errors.New("connection reset by peer"),
```
Comment (vito) [inline retryer.go]: This should be redundant; the `syscall.ECONNRESET` above has the same error message. Did that one not work?

File: retryer.go [hunk 2]
```diff
@@ -38,5 +38,8 @@ var defaultRetryableErrors = []error{
 	syscall.ETIMEDOUT,
 	errors.New("i/o timeout"),
 	errors.New("no such host"),
-	errors.New("remote error: handshake failure"),
+	errors.New("handshake failure"),
```
Comment (vito) [inline retryer.go:41]: Can you give an example of an error that returns this that we'd want to retry? At least with SSH, handshake failure usually means invalid credentials, which isn't retryable.

Can't think of anything else off the top of my head. Willing to pull it in, just curious what the error was.

File: retryer.go [hunk 3]
```diff
@@ -38,5 +38,8 @@ var defaultRetryableErrors = []error{
...
+	errors.New("handshake timeout"),
```
Comment (vito) [inline retryer.go:42]: Makes sense to make this more general - I saw 'TLS handshake timeout' in concourse/concourse#2542.

File: retryer.go [hunk 4]
```diff
@@ -38,5 +38,8 @@ var defaultRetryableErrors = []error{
...
+	errors.New("connection reset by peer"),
+	errors.New("timeout awaiting response headers"),
```
Comment (vito) [inline retryer.go:43]: Makes sense too. Shame `net/http` doesn't export a var for this. Oh well.

**Review by vito — CHANGES_REQUESTED** [review]
Thanks! Just had a few questions inline.

File: retryer.go (hunk 1 above)
Comment (romankartgh) [reply retryer.go]: agreed

File: retryer.go (hunk 3 above)
Comment (romankartgh) [reply retryer.go:42]: yes, exactly

File: retryer.go (hunk 2 above)
Comment (romankartgh) [reply retryer.go:41]: We've seen the following 4 with Vault and Volumes:
```
Finding variable 'consul_token': remote error: tls: handshake failure
Finding variable 'consul_token': Get [variable]: net/http: TLS handshake timeout
Finding variable 'aptomi_docker_registry_password': Get [variable]: dial tcp 13.77.xxx.xx:443: connect: connection reset by peer
Put /volumes/63f08caf-341b-411c-6036-90449cad3d36/stream-in?path=.: net/http: timeout awaiting response headers
```

The theory is that when your service runs on Kubernetes behind Azure LB, GKE LB, etc... these things sometimes happen.

Comment (vito) [reply retryer.go:41]: Thanks. I guess we had it before, too (all the `remote error: handshake failure` Google results are TLS related). I'll go ahead and merge it. We're unlikely to use `retryhttp` for SSH handshaking so it shouldn't be very risky.

**Review by vito — APPROVED** [review]

**Comment by vito** [comment]
thanks! (don't forget to bump `go.mod` in your PR, btw)

**Pull request merged**